From cc6eedbff45df7977ad2d7670fc0194148122059 Mon Sep 17 00:00:00 2001
From: "Dillibabu, Karthikx" <karthikx.dillibabu@intel.com>
Date: Mon, 10 May 2021 13:39:04 +0530
Subject: [PATCH 061/109] FIX for setting autosuspend_delay_ms in SPI issue

HSD : 1509166184
Added pm runtime changes for setting autosuspend_delay_ms.

Signed-off-by: Dillibabu, Karthikx <karthikx.dillibabu@intel.com>
---
 drivers/spi/spi-dw-core.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/spi/spi-dw-core.c b/drivers/spi/spi-dw-core.c
index f7d45318db8a..5191706f0268 100644
--- a/drivers/spi/spi-dw-core.c
+++ b/drivers/spi/spi-dw-core.c
@@ -16,6 +16,7 @@
 #include <linux/spi/spi-mem.h>
 #include <linux/string.h>
 #include <linux/of.h>
+#include <linux/pm_runtime.h>
 
 #include "spi-dw.h"
 
@@ -930,6 +931,11 @@ int dw_spi_add_host(struct device *dev, struct dw_spi *dws)
 		}
 	}
 
+	pm_runtime_set_autosuspend_delay(dev, 50);
+	pm_runtime_use_autosuspend(dev);
+	pm_runtime_set_active(dev);
+	pm_runtime_enable(dev);
+
 	ret = spi_register_controller(master);
 	if (ret) {
 		dev_err(&master->dev, "problem registering spi master\n");
@@ -940,6 +946,9 @@ int dw_spi_add_host(struct device *dev, struct dw_spi *dws)
 	return 0;
 
 err_dma_exit:
+	pm_runtime_put_noidle(dev);
+	pm_runtime_disable(dev);
+
 	if (dws->dma_ops && dws->dma_ops->dma_exit)
 		dws->dma_ops->dma_exit(dws);
 	spi_enable_chip(dws, 0);
-- 
2.25.1

